source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -0,0 +1,306 @@
+   :values: reference
+
+.. meta::
+   :keywords: watch, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use a **change stream** to monitor real-time
+changes to your database. A change stream is a {+mdb-server+} feature that
+allows your application to subscribe to data changes on a collection, database,
+or deployment.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the :ref:`<csharp-quickstart>`.
+
+The examples on this page use the following ``Restaurant``, ``Address``, and ``GradeEntry`` 
+classes as models:
+
+.. literalinclude:: /includes/code-examples/Restaurant.cs
+   :language: csharp
+   :copyable:
+   :dedent:
+
+.. literalinclude:: /includes/code-examples/Address.cs
+   :language: csharp
+   :copyable:
+   :dedent:
+
+.. literalinclude:: /includes/code-examples/GradeEntry.cs
+   :language: csharp
+   :copyable:
+   :dedent:
+
+.. include:: /includes/convention-pack-note.rst
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call the ``Watch()`` or ``WatchAsync()`` method. The instance on which you
+call the method determines the scope of events that the change
+stream listens for. You can call the ``Watch()`` or ``WatchAsync()`` method on the following
+classes:
+
+- ``MongoClient``: To monitor all changes in the MongoDB deployment
+- ``Database``: To monitor changes in all collections in the database
+- ``Collection``: To monitor changes in the collection
+
+The following example opens a change stream on the ``restaurants`` collection
+and outputs the operation type of changes as they occur. Select the
+:guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: change-stream-async
+
+      .. literalinclude:: /includes/code-examples/change-streams/change-streams.cs
+         :start-after: start-open-change-stream-async
+         :end-before: end-open-change-stream-async
+         :language: csharp
+
+   .. tab:: Synchronous
+      :tabid: change-stream-sync
+
+      .. literalinclude:: /includes/code-examples/change-streams/change-streams.cs
+         :start-after: start-open-change-stream
+         :end-before: end-open-change-stream
+         :language: csharp
+
+To begin watching for changes, run the application. Then, in a separate
+application or shell, modify the ``restaurants`` collection. The following
+example updates a document with a ``name`` field value of ``Blarney Castle``:
+
+.. _csharp-change-stream-update:
+
+.. literalinclude:: /includes/code-examples/change-streams/change-streams.cs
+   :start-after: start-modify-document
+   :end-before: end-modify-document
+   :language: csharp
+
+When you update the collection, the change stream application prints the type of change
```

Comment:
S: Since this section is focused on the change stream rather than updating documents, it might be more helpful to include the change stream output produced by modifying the document instead of an example of how to modify a document. [This section](https://www.mongodb.com/docs/drivers/go/current/fundamentals/crud/read-operations/changestream/#example) on the Go page has just the output, for example

Reply:
Updated the code a bit to have a more relevant output and added the output in place of the update code